### PR TITLE
[prometheus] move alertmanager out of prometheus

### PR DESCRIPTION
**Pull request by zanac1986**
#### What this PR does / why we need it

Move node-exporter the embedded chart back to its parent. As mentioned here:

https://github.com/prometheus-community/helm-charts/pull/2714
https://github.com/prometheus-community/helm-charts/pull/2673
https://github.com/prometheus-community/helm-charts/pull/2680
https://github.com/prometheus-community/helm-charts/pull/2627#issuecomment-1300167738
https://github.com/prometheus-community/helm-charts/pull/2627#issuecomment-1300192978
https://github.com/prometheus-community/helm-charts/issues/2654#issuecomment-1306468145

Signed-off-by: zanac1986 <zanhsieh@protonmail.com>

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/values.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ serviceAccounts:
     name:
     annotations: {}
 
-alertmanager:
+old_alertmanager:
```
Comment (monotek) [inline charts/prometheus/values.yaml]: Variablenames sould be camel case.

Why is this needed anyway?

The condition in chart.yaml is "alertmanager.enabled".

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/values.yaml (hunk 1 above)
Comment (zanac1986) [reply charts/prometheus/values.yaml]: @monotek 
There must be some glitch on Github since I can't find `old_alertmanager` in values.yaml.

**Comment by zanac1986** [comment]
For the reason why required Helm 3.7+, see this issue:

https://github.com/prometheus-community/helm-charts/issues/2723

**Review by monotek — APPROVED** [review]
lgtm.

not sure if state / pvc of alertmanager should be migrated too though?

**Comment by zanhsieh** [comment]
> not sure if state / pvc of alertmanager should be migrated too though?
@monotek 
Any tip on how to migrate pvc underneath?

**Comment by monotek** [comment]
> Any tip on how to migrate pvc underneath without data loss?

Could be done via manual steps (kubectl copy from old pvc to local dir and copy from local dir to new pvc).

I started to write a chart for such stuff (with move of prometheus deployment to statefulset in mind) but i need some more time to finish and test it, as i'm occupied with other stuff at the moment.

**Comment by zanac1986** [comment]
> > Any tip on how to migrate pvc underneath without data loss?
> 
> Could be done via manual steps (kubectl copy from old pvc to local dir and copy from local dir to new pvc) but this might mess up permissions.
> 
> I started to write a chart for such stuff (with move of prometheus deployment to statefulset in mind) but i need some more time to finish and test it, as i'm occupied with other stuff at the moment.

Glad to know. Please let me know when I should use it.

**Comment by monotek** [comment]
Like i said.... Don't expect it soon :/

**Review by zanhsieh — APPROVED** [review]

**Comment by ivyostosh** [comment]
@zanac1986 

Sorry I got behind on submitting my PR to move out the alertmanager! I was under the impression some work needs to be done on the alertmanager chart side before doing this change, but it seems to be a misunderstanding on my side. Either way, I would like to ask the following clarification:

I noticed the following templates exist in the old prometheus alertmanager charts, but not in the new alertmanager charts you referenced in this PR:
clusterrole
clusterrolebinding
deploy
headless-svc
netpol
psp
pvc
role
rolebinding

So my questions are twofold:
- What will happen during an upgrade for those templates that don't exist anymore?
- Is there a plan to bring those components back to the alertmanager chart?

**Comment by zanac1986** [comment]
@ivyostosh 
The answer would be somewhat complicated.

Q: What will happen during an upgrade for those templates that don't exist anymore?
A: I think they should get deleted. You could give it a try. So far I didn't see any incompatibility issue yet.

```
helm install prometheus prometheus-community/prometheus --version 15.18.0
helm upgrade prometheus prometheus-community/prometheus --version 18.0.0.0
```

Q: Is there a plan to bring those components back to the alertmanager chart?
A: Would try to bring them back upon requests. For those obsolete (e.g. psp) will get drop. For PVC, I didn't find any good way for migration.

**Comment by monotek** [comment]
@zanac1986 

can you please also check whats the issue after the upgrade here: https://github.com/prometheus-community/helm-charts/issues/2745
